test/integration/consul-container/libs/assert/envoy.go

```diff
@@ -0,0 +1,102 @@
+package assert
+
+import (
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	libservice "github.com/hashicorp/consul/test/integration/consul-container/libs/service"
+	"github.com/hashicorp/consul/test/integration/consul-container/libs/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// GetEnvoyListenerFilters validates that proxy was configured with one rbac listener filter
+func GetEnvoyListenerFilters(t *testing.T, adminPort int) {
+	var (
+		dump string
+		err  error
+	)
+	failer := func() *retry.Timer {
+		return &retry.Timer{Timeout: 30 * time.Second, Wait: 1 * time.Second}
+	}
+
+	retry.RunWith(failer(), t, func(r *retry.R) {
+		dump, err = libservice.GetEnvoyConfigDump(adminPort, "")
+		if err != nil {
+			r.Fatal("could not fetch envoy configuration")
+		}
+	})
+	// Make sure there is one rbac listener listener configured
+	filter := `.configs[2].dynamic_listeners[].active_state.listener | "\(.name) \( .filter_chains[0].filters | map(.name))"`
+	results, err := utils.JQFilter(dump, filter)
+	if err != nil {
+		t.Fatal("could not parse envoy configuration")
+	}
+
+	if len(results) != 2 {
+		t.Fatalf("s1 proxy should have been configured with one rbac listener filter. Got %d listener(s)", len(results))
+	}
+
+	// validate public listeners value
```

Comment:
Could you elaborate a bit on why the following code  validates public listeners value? Thanks.